### ISSUE TITLE
Response in API or Service cause error

### ISSUE THREAD

**Issue by jefferai** [goadesign/goa#2473]
The docs say:
> Response may appear in an API or service expression to define error responses common to all the API or service methods.

But when I put a Response in an API block I get an error `invalid use of Response in API`. A similar thing happens if I put it in Service.

**Comment by tchssk** [comment]
Do you use `Response` in `HTTP` or `GRPC` ?

```go
// invalid use
var _ = API(func() {
	Response(StatusOK)
	Response(CodeOK)
})

var _ = Service("foo", func() {
	Response(StatusOK)
	Response(CodeOK)
}

// valid use
var _ = API(func() {
	HTTP(func() {
		Response(StatusOK)
	})
	GRPC(func() {
		Response(CodeOK)
	})
})

var _ = Service("foo", func() {
	HTTP(func() {
		Response(StatusOK)
	})
	GRPC(func() {
		Response(CodeOK)
	})
}
```

**Comment by jefferai** [comment]
I was definitely using it int he invalid way. In my defense, as a newbie, the docs _dont'_ say that they can be used in Service or API _in an HTTP or GRPC_ block. :-) Might be worth clarifying in the docs.

**Comment by raphael** [comment]
@jefferai I'd be happy to merge a PR that makes the docs clearer!

**Comment by jefferai** [comment]
Reopening because it doesn't work the way you suggested either. When I move Response into HTTP/GRPC functions in the API declaration:
```
var _ = API("controller", func() {
        Version("0.1")
        Server("controller", func() {
                Host("development", func() {
                        Description("Development default host")
                        URI("http://localhost:9200")
                        URI("grpc://localhost:9200")
                })
        })
        HTTP(func() {
                Response("forbidden", StatusForbidden)
        })
        GRPC(func() {
                Response("forbidden", CodePermissionDenied)
        })
})
```
I get error messages:
```
exit status 1
[../../go/pkg/mod/goa.design/goa/v3@v3.0.9/dsl/response.go:163] invalid use of Response in design
[../../go/pkg/mod/goa.design/goa/v3@v3.0.9/dsl/grpc.go:103] invalid use of GRPC in API controller
```
If I comment those out or remove them the errors go away.

**Comment by tchssk** [comment]
The godoc for `GRPC` says

> GRPC defines gRPC transport specific properties on an API, a service, or a single method.

but the implementation doesn't support `API`. Is it a bug? @raphael

**Comment by raphael** [comment]
We went a little bit back and forth on this as we developed v2 but where we landed is:
* Errors can be defined in services or methods.
* The corresponding transport mappings can thus also be defined in services and methods.
* The only transport specific concern that can be defined at the API level is the HTTP base path.

We should update the docs to reflect the above. We can then open a separate feature request for defining errors and transport mappings at the API level if that's something that we'd like to add. Thank you for raising the issue!

**Comment by jefferai** [comment]
Not stale, just waiting on the discussed update.

**Comment by raphael** [comment]
@jefferai thank you for the reminder! I have updated the comment to reflect the current behavior properly. Consider opening a separate issue for making it possible to define errors and responses at the API level. Note that errors and responses defined at the "global" levels (API or Service) must apply to all the API/service endpoints. This is a little bit different from the semantic of OpenAPI for example that makes it possible to define common responses at the API level as a set of "pre-ready" responses that can then be used in endpoints. This is the main reason for disallowing these at the API level in Goa as it could be confusing to users that expect the OpenAPI behavior.

**Comment by jefferai** [comment]
Gotcha, thanks!
